core/block_render_svg_vertical.js

```diff
@@ -974,6 +976,55 @@ Blockly.BlockSvg.prototype.renderDraw_ = function(iconWidth, inputRows) {
 };
 
 /**
+ * Give the block an attribute 'data-shapes' that lists its shape[s]
+ * @private
+ */
+Blockly.BlockSvg.prototype.renderClassify_ = function() {
+  var shapes = [];
+  
+  if(this.outputConnection) {
+    if(this.isShadow_) {
+      shapes.push("input");
+    } else {
+      shapes.push("reporter");
+    }
+    if(this.edgeShape_ === Blockly.OUTPUT_SHAPE_HEXAGONAL) {
+      shapes.push("boolean");
+    } else if(this.edgeShape_ === Blockly.OUTPUT_SHAPE_ROUND) {
+      shapes.push("string");
+    } else {
+      // this shouldn't happen anymore because string-shaped reporters were phased out
+    }
+  } else {
+    // count the number of statement inputs
+    var inputList = this.inputList;
+    var statementCount = 0;
+    for (var i = 0, input; input = inputList[i]; i++) {
+      if (input.connection && input.connection.type === Blockly.NEXT_STATEMENT) {
+        statementCount++;
+      }
+    }
+    
+    if(statementCount) {
+      shapes.push("c-block");
+      if(statementCount > 1) {
+        shapes.push("else");
+      }
```

Comment:
I like that idea a lot. I'll leave the "c-block" shape in as well since I worry that it'll make it harder to detect any c-block with CSS if we just use the numbers